### ISSUE TITLE
generate server & client for  http://nim-lang.org/

### ISSUE THREAD

**Issue by despiegk** [Jumpscale/go-raml#158]
- generate for nim-lang
- make sure basic validation happens
- make sure REST UI is inside (easy exploration)
- write code to test performance
- make sure there is an example

phase 2
- also support SSL & oauth2

**Comment by iwanbk** [comment]
I can only find `nawak` and `jester`. While `nawak` got better techempower benchmark result, it seems not maintained anymore. last update time is 2 years ago.

I think `jester` is better choice.

**Comment by despiegk** [comment]
did we do a perf test? if yes how good was it?
how far are we with code?

**Comment by iwanbk** [comment]
```
did we do a perf test? if yes how good was it?
```
not yet

```
how far are we with code?
```
basic server & client has been finished. i am writing the tutorial now, should be finished in less than an hour.

perf test will be done after the tutorial

**Comment by despiegk** [comment]
cant find tutorial

**Comment by despiegk** [comment]
do example with jester

**Comment by iwanbk** [comment]
It is in nim-server branch, docs/tutorial/nim folder. Still not merged, because still has issue with travis,

**Comment by despiegk** [comment]
when can we merge it in?

**Comment by despiegk** [comment]
is there tutorial how to integrate authentication/authorization with itsyou.online for nim rest webserver

**Comment by iwanbk** [comment]
```
when can we merge it in?
```
Already merged

```
is there tutorial how to integrate authentication/authorization with itsyou.online for nim rest webserver
```
no integration with itsyou.online yet

**Comment by iwanbk** [comment]
about oauth2/itsyouonline support, our problem is : jester doesn't support middleware.

We also still don't have input validation, because there is still no library for it

**Comment by iwanbk** [comment]
```
jester doesn't support middleware.
```
i'll see if we can use Nim `proc` or `template` for this

**Comment by iwanbk** [comment]
oauth2 has been implemented in PR #198.

About the SSL,  jester unfortunately has no support for SSL.

Same case with Nim's httpserver and asynchttpserver. Not only doesn't support SSL, the docs said that it is not production ready.
So, best solution is using proxy like nginx or haproxy.
